### PR TITLE
issue #33 - s2697, ta 13019: store MD5 checksum on PUT for fs adapter

### PR DESCRIPTION
**Pull request by hjeeves**
Cleaned up some unused imports as well. Added 'list' to the get-put-delete test to make sure the iterator returns a value in the checksum. More than that can't be done until the iterator is fixed (issue #66 )

### CODE REVIEW

File: cadc-storage-adapter-fs/src/main/java/org/opencadc/inventory/storage/fs/FileSystemStorageAdapter.java [hunk 1]
```diff
@@ -582,5 +590,18 @@ private Path createStorageLocationPath(StorageLocation storageLocation) {
         Path ret = root.resolve(path.toString());
         return ret;
     }
-    
+
+    public static void setFileAttribute(Path path, String propertyName, String property) throws IOException {
```
Comment (pdowler) [inline cadc-storage-adapter-fs/src/main/java/org/opencadc/inventory/storage/fs/FileSystemStorageAdapter.java]: would prefer args that was key-value or name-value pair... "property" is often sometimes used to indicate the key but usually means the pair (key and value)

line ~599: calling getBytes(Charset) twice, but I don't see the value if logging the bytes when you already logged the string key = value

This method only needs one log.debug with all three method args - spreading it over two lines loses information.

And finally some ultra-picky optimisation: you can move then view creation inside the if statement

File: cadc-storage-adapter-fs/src/main/java/org/opencadc/inventory/storage/fs/FileSystemIterator.java [hunk 2]
```diff
@@ -100,6 +98,7 @@
     private PathItem next = null;
     Stack<StackItem> stack;
     private String fixedParentDir;
+    public static final String CHECKSUM_ATTRIBUTE_NAME = "Artifact.contentChecksum";
```
[the remark this answers is not in the document]
Comment (hjeeves) [reply cadc-storage-adapter-fs/src/main/java/org/opencadc/inventory/storage/fs/FileSystemIterator.java]: ok

File: cadc-storage-adapter-fs/src/main/java/org/opencadc/inventory/storage/fs/FileSystemStorageAdapter.java [hunk 3]
```diff
@@ -136,6 +140,7 @@
     public static final String CONFIG_PROPERTY_ROOT = "root";
     public static final String CONFIG_PROPERTY_BUCKETMODE = "bucketMode";
     public static final String CONFIG_PROPERTY_BUCKETDEPTH = "bucketLength";
+    public static final String CHECKSUM_ATTRIBUTE_NAME = "Artifact.contentChecksum";
```
[the remark this answers is not in the document]
Comment (hjeeves) [reply cadc-storage-adapter-fs/src/main/java/org/opencadc/inventory/storage/fs/FileSystemStorageAdapter.java]: hope I got that right...

File: cadc-storage-adapter-fs/src/main/java/org/opencadc/inventory/storage/fs/FileSystemStorageAdapter.java (hunk 1 above)
Comment (hjeeves) [reply cadc-storage-adapter-fs/src/main/java/org/opencadc/inventory/storage/fs/FileSystemStorageAdapter.java]: done

**Pull request merged**